packages/common/src/controllers/CheckoutController.ts

```diff
@@ -76,19 +79,30 @@ export default class CheckoutController {
       couponCode,
     };
 
-    const response = await this.checkoutService.updateOrder(updateOrderPayload);
-    if (response.errors.length > 0) {
-      // clear the order when the order doesn't exist on the server
-      if (response.errors[0].includes(`Order with ${order.id} not found`)) {
-        useCheckoutStore.getState().setOrder(null);
+    try {
+      const response = await this.checkoutService.updateOrder(updateOrderPayload);
+
+      if (response.errors.length > 0) {
+        // clear the order when the order doesn't exist on the server
+        if (response.errors[0].includes(`Order with ${order.id} not found`)) {
+          useCheckoutStore.getState().setOrder(null);
+        }
+
+        throw new FormValidationError({ order: [response.errors[0]] });
```

Comment:
You have space, why not keep all the errors?
```suggestion
        throw new FormValidationError({ order: response.errors });
```

Reply:
We only show the first error on all form fields and want to remove the `string[]` type. But if not, we could pass all errors indeed.